### PR TITLE
VP-2291: Rewrite export according to V3 pattern

### PR DESCRIPTION
**Pull request by AliveMen**
Fix broken export

### CODE REVIEW

File: src/VirtoCommerce.CatalogPersonalizationModule.Web/ExportImport/PersonalizationExportImport.cs [hunk 1]
```diff
@@ -1,60 +1,94 @@
...
-            progressCallback(progressInfo);
+                var taggedItems = await _taggedItemSearchService.SearchTaggedItemsAsync(new TaggedItemSearchCriteria {Take = int.MaxValue});
+                progressInfo.Description = $"{taggedItems.TotalCount} tagged items loading...";
+                progressCallback(progressInfo);
 
-            await _taggedItemService.SaveChangesAsync(backupObject.TaggedItems.ToArray());
+                await writer.WritePropertyNameAsync("TaggedItemsTotalCount");
+                await writer.WriteValueAsync(taggedItems.TotalCount);
+
+                await writer.WritePropertyNameAsync("TaggedItems");
```
Comment (yecli) [inline src/VirtoCommerce.CatalogPersonalizationModule.Web/ExportImport/PersonalizationExportImport.cs:47]: Could you use paged generic methods like here, to not read all data at one query?
https://github.com/VirtoCommerce/vc-module-catalog/blob/dev/src/VirtoCommerce.CatalogModule.Data/ExportImport/CatalogExportImport.cs#L71

File: src/VirtoCommerce.CatalogPersonalizationModule.Web/ExportImport/PersonalizationExportImport.cs [hunk 2]
```diff
@@ -1,60 +1,94 @@
...
+                await writer.WriteStartObjectAsync();
 
-            progressInfo.Description = String.Format("{0} tagged items importing...", backupObject.TaggedItems.Count);
-            progressCallback(progressInfo);
+                var taggedItems = await _taggedItemSearchService.SearchTaggedItemsAsync(new TaggedItemSearchCriteria {Take = int.MaxValue});
+                progressInfo.Description = $"{taggedItems.TotalCount} tagged items loading...";
+                progressCallback(progressInfo);
 
-            await _taggedItemService.SaveChangesAsync(backupObject.TaggedItems.ToArray());
+                await writer.WritePropertyNameAsync("TaggedItemsTotalCount");
```
Comment (yecli) [inline src/VirtoCommerce.CatalogPersonalizationModule.Web/ExportImport/PersonalizationExportImport.cs:44]: Probably we do not need this count

Comment (AliveMen) [reply src/VirtoCommerce.CatalogPersonalizationModule.Web/ExportImport/PersonalizationExportImport.cs]: removed

File: src/VirtoCommerce.CatalogPersonalizationModule.Web/ExportImport/PersonalizationExportImport.cs (hunk 1 above)
Comment (AliveMen) [reply src/VirtoCommerce.CatalogPersonalizationModule.Web/ExportImport/PersonalizationExportImport.cs:41]: updated

**Pull request merged**